src/Microsoft.DotNet.Cli.Utils/CommandResolution/ToolPathCalculator.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Collections.Generic;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class ToolPathCalculator
+    {
+        private readonly string _packagesDirectory;
+
+        public ToolPathCalculator(string packagesDirectory)
+        {
+            _packagesDirectory = packagesDirectory;
+        }
+
+        public string GetBestLockFilePath(string packageId, VersionRange versionRange, NuGetFramework framework)
```

Comment:
cc @joelverhagen 

Without the tool baked into the consumer app lockfile yet I needed to put this in place. 
In the long term I'd like to use NuGet's libraries for this.